Src/BootCamp.Chapter/FullCommand.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+namespace BootCamp.Chapter
+{
+    internal class FullCommand
+    {
+        public FullCommand()
+        {
+        }
+
+        public EventHandler<FilterCompletedEventArgs> FilterCompleted;
+
+        internal void ExecuteCommand(Stream stream)
+        {
+            using var transactionStream = new TransactionStream(stream);
+            var transactions = transactionStream.ReadTransactionUntilEnd().GroupBy(x => x.ShopName).ToArray(); 
+
+            foreach (var shop in transactions)
+            {
+                OnFilterCompleted(shop);
+            }
+        }
+
+        private void OnFilterCompleted(IGrouping<string, Transaction> shop)
+        {
+            var local = FilterCompleted;
```

Comment:
Why is there a local? not needed and ``FilterCompleted`` can be invoked direclty.

Reply:
[How to publish events that conform to .NET Framework Guidelines (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines#example)

Note the implementation of `OnRaiseCustomEvent`, 

    // Make a temporary copy of the event to avoid possibility of
    // a race condition if the last subscriber unsubscribes
    // immediately after the null check and before the event is raised.